### PR TITLE
cleanup accesses to subtype() and subtypes()

### PR DESCRIPTION
**Pull request by kroening**
- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [x] My contribution is formatted in line with CODING_STANDARD.md.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- [X] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: src/util/type.h [hunk 1]
```diff
@@ -141,6 +141,18 @@ class type_with_subtypet:public typet
   #endif
 };
 
+const type_with_subtypet &to_type_with_subtype(const typet &type)
```
Comment (tautschnig) [inline src/util/type.h]: Here and all other newly introduced functions here: missing `inline`

**Review by tautschnig — CHANGES_REQUESTED** [review]

**Comment by tautschnig** [comment]
Also needs a rebase on current develop to make CI pass.

File: src/util/type.h (hunk 1 above)
Comment (kroening) [reply src/util/type.h]: Done

**Review by tautschnig — APPROVED** [review]
Approving modulo CI passing.

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 02a1b95a3b411df03c1fdd3063e5b96323a1f125).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/85397997

**Pull request merged**